src/credentials/StaticGroupDataProvider.cpp

```diff
@@ -0,0 +1,780 @@
+    {
+        return GetFabric(fabric_index, /* allow_allocate= */ true);
+    }
+
+public:
+    CHIP_ERROR Init() override
+    {
+        // Clear-out all entries of index mapping.
+        for (size_t i = 0; i < kMaxNumGroupStates; ++i)
+        {
+            mGroupStateIndices[i].Invalidate();
+        }
+        mInitialized = true;
+        return CHIP_NO_ERROR;
+    }
+
+    void Finish() override { mInitialized = false; }
+
+    // Endpoints
+    bool GroupMappingExists(chip::FabricIndex fabric_index, GroupMapping & mapping) override
+    {
+        VerifyOrReturnError(mInitialized, false);
+
+        Fabric * fabric = GetExistingFabric(fabric_index);
+        VerifyOrReturnError(fabric, false);
+
+        for (uint16_t i = 0; fabric && i < kEndpointEntriesMax; ++i)
+        {
+            EndpointEntry & entry = fabric->endpoints[i];
+            if (entry.in_use && (entry == mapping))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    CHIP_ERROR AddGroupMapping(chip::FabricIndex fabric_index, GroupMapping & mapping, const char * name) override
+    {
+        VerifyOrReturnError(mInitialized, CHIP_ERROR_INTERNAL);
+
+        (void) name; // Unused!
+
+        Fabric * fabric = GetExistingFabricOrAllocateNew(fabric_index);
+        VerifyOrReturnError(fabric, CHIP_ERROR_NO_MEMORY);
+
+        // Search for existing mapping
+        for (uint16_t i = 0; i < kEndpointEntriesMax; ++i)
+        {
+            EndpointEntry & entry = fabric->endpoints[i];
+            if (entry.in_use && (entry == mapping))
+            {
+                // Duplicated
+                return CHIP_NO_ERROR;
+            }
+        }
+        // New mapping
+        for (uint16_t i = 0; i < kEndpointEntriesMax; ++i)
+        {
+            EndpointEntry & entry = fabric->endpoints[i];
+            if (!entry.in_use)
+            {
+                entry.group    = mapping.group;
+                entry.endpoint = mapping.endpoint;
+                entry.in_use   = true;
+                return CHIP_NO_ERROR;
+            }
+        }
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    CHIP_ERROR RemoveGroupMapping(chip::FabricIndex fabric_index, GroupMapping & mapping) override
+    {
+        VerifyOrReturnError(mInitialized, CHIP_ERROR_INTERNAL);
+
+        Fabric * fabric = GetExistingFabric(fabric_index);
+        VerifyOrReturnError(fabric, CHIP_ERROR_INVALID_FABRIC_ID);
+
+        // Search for existing mapping
+        for (uint16_t i = 0; fabric && i < kEndpointEntriesMax; ++i)
+        {
+            EndpointEntry & entry = fabric->endpoints[i];
+            if (entry.in_use && (entry == mapping))
+            {
+                // Found
+                entry.in_use = false;
+                return CHIP_NO_ERROR;
+            }
+        }
+
+        return CHIP_ERROR_KEY_NOT_FOUND;
+    }
+
+    CHIP_ERROR RemoveAllGroupMappings(chip::FabricIndex fabric_index, EndpointId endpoint) override
```

Comment:
endpoint argument is unused. Suggest removing.

Reply:
Argument removed.